core/src/nucleus/ribosome/api/get_links.rs

```diff
@@ -16,47 +15,132 @@ pub fn invoke_get_links(
 ) -> Result<Option<RuntimeValue>, Trap> {
     // deserialize args
     let args_str = runtime.load_utf8_from_args(&args);
-    let res_entry: Result<GetLinksArgs, _> = serde_json::from_str(&args_str);
-    // Exit on error
-    if res_entry.is_err() {
-        return ribosome_error_code!(ArgumentDeserializationFailed);
+    let input: GetLinksArgs = match serde_json::from_str(&args_str) {
+        Ok(input) => input,
+        Err(_) => return ribosome_error_code!(ArgumentDeserializationFailed),
+    };
+
+    let get_links_result = runtime
+        .context
+        .state()
+        .unwrap()
+        .dht()
+        .get_links(input.entry_address, input.tag);
+
+    let json = serde_json::to_string(&GetLinksResult {
+        ok: get_links_result.is_ok(),
+        links: get_links_result
+            .clone()
+            .unwrap_or(HashSet::new())
+            .iter()
+            .map(|eav| eav.value())
+            .collect::<Vec<Address>>(),
+        error: get_links_result
+            .map_err(|holochain_error| holochain_error.to_string())
+            .err()
+            .unwrap_or(String::from("")),
```

Comment:
Note: With upcoming error system overhaul, we won't have to do this anymore and will have a standard way to send back an error, instead of putting it in the normal result struct.

Reply:
I prefer this explicit way actually. The results of these functions need to be understood by every HDK that will follow in the future. I would like to make it easily understandable by people not deep into Rust.

`ribosome_error_code!` above fooled me today as it does return an `Ok( <something> )`. Since everything that this function returns gets serialized into a string that then the HDK sees, returning special strings that are not listed in `wasm_utils::api_serialization` might confuse people. (It did confuse me already)